src/backend/storeManagers/gog/games.ts

```diff
@@ -882,3 +894,157 @@ export function isGameAvailable(appName: string) {
   }
   return false
 }
+
+async function postPlaytimeSession({
+  appName,
+  session_date,
+  time
+}: GOGSessionSyncQueueItem) {
+  const userData: UserData | undefined = configStore.get_nodefault('userData')
+
+  const credentials = await GOGUser.getCredentials().catch(() => null)
+
+  if (!userData || !credentials) {
+    logError(
+      "No userData in config or couldn't fetch credentials, unable to post new session",
+      {
+        prefix: LogPrefix.Gog
+      }
+    )
+    return null
+  }
+
+  return axios
+    .post(
+      `https://gameplay.gog.com/games/${appName}/users/${userData?.galaxyUserId}/sessions`,
+      { session_date, time },
+      {
+        headers: {
+          Authorization: `Bearer ${credentials.access_token}`
+        }
+      }
+    )
+    .catch(() => {
+      return null
+    })
+}
+
+export async function updateGOGPlaytime(
+  appName: string,
+  startPlayingDate: Date,
+  finishedPlayingDate: Date
+) {
+  // Let server know about new session
+  const sessionDate = Math.floor(startPlayingDate.getTime() / 1000) // In seconds
+  const time = Math.floor(
+    (finishedPlayingDate.getTime() - startPlayingDate.getTime()) / 1000 / 60
+  ) // In minutes
+
+  // It makes no sense to post 0 minutes of playtime
+  if (time < 1) {
+    return
+  }
+
+  const data = {
+    session_date: sessionDate,
+    time
+  }
+  const userData: UserData | undefined = configStore.get_nodefault('userData')
+
+  if (!userData) {
+    logWarning(['Unable to post session, userData not present'], {
+      prefix: LogPrefix.Gog
+    })
+    return
+  }
+
+  if (!isOnline()) {
+    logWarning(['App offline, unable to post new session at this time'], {
+      prefix: LogPrefix.Gog
+    })
+    const alreadySetData = playtimeSyncQueue.get(userData.galaxyUserId, [])
+    alreadySetData.push({ ...data, appName })
+    playtimeSyncQueue.set(userData.galaxyUserId, alreadySetData)
+    runOnceWhenOnline(syncQueuedPlaytimeGOG)
+    return
+  }
+
+  const response = await postPlaytimeSession({ ...data, appName })
+
+  if (!response || response.status !== 201) {
+    logError('Failed to post session', { prefix: LogPrefix.Gog })
+    const alreadySetData = playtimeSyncQueue.get(userData.galaxyUserId, [])
+    alreadySetData.push({ ...data, appName })
+    playtimeSyncQueue.set(userData.galaxyUserId, alreadySetData)
+    return
+  }
+
+  logInfo('Posted session to gameplay.gog.com', { prefix: LogPrefix.Gog })
+}
+
+export async function syncQueuedPlaytimeGOG() {
+  if (playtimeSyncQueue.has('lock')) {
+    return
+  }
+  playtimeSyncQueue.set('lock', [])
+  const userData: UserData | undefined = configStore.get_nodefault('userData')
+  if (!userData) {
+    logError('Unable to syncQueued playtime, userData not present', {
+      prefix: LogPrefix.Gog
+    })
+    return
+  }
+  const queue = playtimeSyncQueue.get(userData.galaxyUserId, [])
+  const failed = []
+
+  for (const session of queue) {
+    if (!isOnline) {
```

Comment:
```suggestion
    if (!isOnline()) {
```

this is the imported function, right?